src/chronify/time_series_mapper_representative.py

```diff
@@ -0,0 +1,140 @@
+                    print(e)
+                    print(f"check_timestamps failed, dropping mapped table {self._to_schema.name}")
+                    conn.execute(text(f"DROP TABLE {self._to_schema.name}"))
+        finally:
+            with self._engine.connect() as conn:
+                conn.execute(text(f"DROP TABLE {map_table_name}"))
+
+    def _create_mapping(self, is_tz_naive) -> pd.DataFrame:
+        """Create mapping dataframe"""
+        timestamp_generator = make_time_range_generator(self._to_schema.time_config)
+
+        to_time_col = self._to_schema.time_config.time_column
+        dft = pd.Series(timestamp_generator.list_timestamps()).rename(to_time_col).to_frame()
+        if is_tz_naive:
+            dfm = self._generator.create_tz_naive_mapping_dataframe(dft, to_time_col)
+        else:
+            with self._engine.connect() as conn:
+                table = Table(self._from_schema.name, self._metadata)
+                stmt = (
+                    select(table.c["time_zone"])
+                    .distinct()
+                    .where(table.c["time_zone"].is_not(None))
+                )
+                time_zones = read_database(stmt, conn, self._from_schema.time_config)[
+                    "time_zone"
+                ].to_list()
+            dfm = self._generator.create_tz_aware_mapping_dataframe(dft, to_time_col, time_zones)
+        return dfm
+
+    def _apply_mapping(self, map_table_name: str):
+        """Apply mapping to create result as a table according to_schema"""
+        left_table = Table(self._from_schema.name, self._metadata)
+        right_table = Table(map_table_name, self._metadata)
+        left_table_columns = [x.name for x in left_table.columns]
+        right_table_columns = [x.name for x in right_table.columns]
+
+        final_cols = set(self._to_schema.list_columns())
+        left_cols = set(left_table_columns).intersection(final_cols)
+        right_cols = final_cols - left_cols
+
+        select_stmt = [left_table.c[x] for x in left_cols]
+        select_stmt += [right_table.c[x] for x in right_cols]
+
+        keys = self._from_schema.time_config.list_time_columns()
+        if not self._to_schema.time_config.is_time_zone_naive():
+            keys.append("time_zone")
+            assert (
+                "time_zone" in left_table_columns
+            ), f"time_zone not in table={self._from_schema.name}"
+            assert "time_zone" in right_table_columns, f"time_zone not in table={map_table_name}"
+        on_stmt = ()
+        for i, x in enumerate(keys):
+            if i == 0:
+                on_stmt = left_table.c[x] == right_table.c[x]
+            else:
+                on_stmt &= left_table.c[x] == right_table.c[x]
+        query = select(*select_stmt).select_from(left_table).join(right_table, on_stmt)
+        create_table(self._to_schema.name, query, self._engine, self._metadata)
+        self._metadata.reflect(self._engine, views=True)
```

Comment:
This is already happening in create_table, right? (and can be deleted)

Reply:
I think it's more complete for it to exists here. We should always update the metadata within the same function after creating something.

Reply:
It's not a cheap function, but it's time is nothing compared to the time to do this mapping. Still, though, I would prefer that we minimize the number of times it's called. If we end up calling create_table in 15 places, I really hope that every call site doesn't repeat this behavior. I think it would be fine to document in create_table that it takes care of the update. Also, I'm getting warning errors from duckdb_engine every time it's called.

Reply:
This is the last question. Since `create_table` has taken on the responsibility of updating the metadata, can this line be removed?